main.py

```diff
@@ -118,6 +117,16 @@ def debug_gql(self) -> int:
         "--debug-gql", dest="_debug_gql", action="store_true", help=argparse.SUPPRESS
     )
     args = parser.parse_args(namespace=ParsedArgs())
+
+    if not args.cli:
+        root = tk.Tk()
+        root.overrideredirect(True)
+        root.withdraw()
+        root.iconphoto(
+            True, PhotoImage(master=root, image=Image_module.open(resource_path("pickaxe.ico")))
+        )
+        root.update()
+
```

Comment:
You can't just move the dummy root window after the parser, because `Parser.exit` calls `messagebox.showerror` (L50) which uses the created window to display the error message, thus the window has to exist already, before `parse_args` is called. To resolve this dependency loop, one would need to restructure the `Parser` so that it returns the error message, or `None` if there wasn't one, and then one should check after `parse_args` to see if an error message was returned, then display it accordingly. This would need to be done while preserving the original argparse interface, and possibly without using global variables though, so it's tricky to implement.

Reply:
It seems a nice solution.

But another problem is that in a display-less environment, **creating** a window will cause an error, so one would need to know whether CLI mode is enabled before creating the window. That's why I move the parser before the dummy root window to get the `--cli` argument. (Is it a good way to wrap the code creating the window with a `try`?)